include/core/utilities.h

```diff
@@ -489,24 +489,96 @@ int
 get_max_number_of_boundary_conditions(const std::string &file_name);
 
 /**
- * @brief Return the tensor corresponding to the entry (@p entry_string). If
- * the dimension correspondence of the @p entry_string is not equivalent to
+ * @brief Return the tensor corresponding to the @p value_string. If the
+ * dimension correspondence of the @p value_string is not equivalent to
  * @p spacedim (either 2 or 3), an exception will be thrown. The delimiter
- * separating the elements
- * of the @p entry_string is a comma (",").
+ * separating the elements of the @p value_string is a comma (",").
  *
  * @remark The function can be use to construct Point<spacedim> objects.
  *
  * @tparam spacedim Number of spatial dimensions (2D or 3D).
  *
- * @param[in] entry_string A string in the parameter file.
+ * @param[in] value_string A string in the parameter file corresponding to a
+ * given tensor.
  *
- * @return A Tensor<1,spacedim> corresponding to the @p entry_string in the
+ * @return A Tensor<1,spacedim> corresponding to the @p value_string in the
  * parameter file.
  */
+template <int spacedim>
+inline Tensor<1, spacedim>
+value_string_to_tensor(const std::string &value_string)
+{
+  std::vector<std::string> vector_of_string(
+    Utilities::split_string_list(value_string));
+  std::vector<double> vector_of_double =
+    Utilities::string_to_double(vector_of_string);
+
+  AssertThrow(vector_of_double.size() == 3 || vector_of_double.size() == 2,
```

Comment:
```suggestion
  AssertThrow(vector_of_double.size() == spacedim,
```

Reply:
From what I've seen in the code (tests), sometime people leave the string at 2 components, when they want a 3 component rank 1 tensor also.
That's why I left it general. 
(same for other instances)